### DISCUSSION TITLE
BRDF estimation question

### DISCUSSION THREAD

**Discussion by Jackson334-prog** [mitsuba-renderer/mitsuba3#1018] [Q&A] [answered]
Hi I'm here. I try to setimate the surface material, and I use the plugin `normal` and `principled` BSDFs your work implementation. And Now I estimate the albedo, roughness , normal and metallic map of the surface all conbine them together to estimate using inverse rendering. 
But I was confused about if I conbine them all together, they seem look quite a strange **they will mixed together, eg. normal map contain the base color**.
So I asked for your help , would you kindly like to share with me how to estimate solve my problem, I guess I can change the learning procedure into two stages , the first differentiate the **albedo** and the second differentiate the **other**?
the following was the surface map I have estimated: you can see all map contain the albedo location (1,0). (0,0) was the normal map (0,1) and (1,1) was the metallic and roughness.
![图片1](https://github.com/mitsuba-renderer/mitsuba3/assets/73884949/dd120f6c-0cec-4499-900f-89383dd5a09c)

**Comment by njroussel** [comment, marked as answer]
Hi @ljijjiang1234 

I think that this is simply due to the ambiguity of the optimization task you're trying to solve. For example, in order to make some part of a surface darker, the optimizer has many choices: it can darken the albedo, it can change the roughness, it can move the normal to receive less incident light, ...

Even if you split the optimization into multiple stages, you're at the risk of the earlier stages learning features that should only be learnt by later stages.

This is fundamentally just a hard problem, and you'd need to extra information to disambiguate the problem or regularize your textures for albedo/normals/etc.
